src/main/java/com/cburch/logisim/std/ttl/Ttl74192.java

```diff
@@ -0,0 +1,244 @@
+      "Data Output B",
+      "Data Output A",
+      "Count Down",
+      "Count Up",
+      "Data Output C",
+      "Data Output D",
+      "Data Input D",
+      "Data Input C",
+      "Load",
+      "Carry",
+      "Borrow",
+      "Clear",
+      "Data Input A"
+  };
+  private static final byte[] OUTPUT_PORTS = {2, 3, 6, 7, 12, 13};
+
+  private final int maxVal;
+
+  private Value cntD_prev = Value.NIL;
+  private Value cntU_prev = Value.NIL;
+
+  public Ttl74192() {
+    super(_ID, (byte) 16, OUTPUT_PORTS, PORT_NAMES, null);
+    super.setInstancePoker(Poker.class);
+    maxVal = 9;
+  }
+
+  public Ttl74192(String name, int maxVal) {
+    super(name, (byte) 16, OUTPUT_PORTS, PORT_NAMES, null);
+    super.setInstancePoker(Poker.class);
+    this.maxVal = maxVal;
+  }
+
+  public static class Poker extends InstancePoker {
+    boolean isPressed = true;
+
+    private boolean isInside(InstanceState state, MouseEvent e) {
+      final var p = getTranslatedTtlXY(state, e);
+      var inside = false;
+      for (var i = 0; i < 4; i++) {
+        final var dx = p.x - (56 + i * 10);
+        final var dy = p.y - 30;
+        final var d2 = dx * dx + dy * dy;
+        inside |= (d2 < 4 * 4);
+      }
+      return inside;
+    }
+
+    private int getIndex(InstanceState state, MouseEvent e) {
+      final var p = getTranslatedTtlXY(state, e);
+      for (var i = 0; i < 4; i++) {
+        int dx = p.x - (56 + i * 10);
+        int dy = p.y - 30;
+        int d2 = dx * dx + dy * dy;
+        if (d2 < 4 * 4) return 3 - i;
+      }
+      return 0;
+    }
+
+    @Override
+    public void mousePressed(InstanceState state, MouseEvent e) {
+      isPressed = isInside(state, e);
+    }
+
+    @Override
+    public void mouseReleased(InstanceState state, MouseEvent e) {
+      if (!state.getAttributeValue(TtlLibrary.DRAW_INTERNAL_STRUCTURE).booleanValue()) return;
+      if (isPressed && isInside(state, e)) {
+        var index = getIndex(state, e);
+        final var data = (TtlRegisterData) state.getData();
+        if (data == null) return;
+        var current = data.getValue().toLongValue();
+        final long bitValue = 1 << index;
+        current ^= bitValue;
+        updateState(state, current, Value.FALSE, Value.FALSE);
+      }
+      isPressed = false;
+    }
+  }
+
+  @Override
+  public void paintInternal(InstancePainter painter, int x, int y, int height, boolean up) {
+    final var gfx = (Graphics2D) painter.getGraphics();
+    super.paintBase(painter, false, false);
+    Drawgates.paintPortNames(
+        painter,
+        x,
+        y,
+        height,
+        new String[] {
+          "B", "QB", "QA", "CntD", "CntU", "QC", "QD", "D", "C", "LOAD", "CAR", "BOR", "CLR", "A"
+        });
+    TtlRegisterData data = (TtlRegisterData) painter.getData();
+    drawState(gfx, x, y, height, data);
+  }
+
+  private void drawState(Graphics2D gfx, int x, int y, int height, TtlRegisterData state) {
+    if (state != null) {
```

Comment:
Can I request small refactoring. All the places where you do 

```java
if (state != null) {
     ... tons of lines
}
```

replace with

```java

if (state == null) return;
... tons of lines
```

whenever applicable (a few places in your current PR). That reduces "brace tree" across the code.